.github/workflows/check-source-signatures.yml

```
@@ -45,9 +45,25 @@ jobs:
       - name: Get changed packages
         run: |
           # Find the packages that have been modified in the current PR. They will be of the form '/path/to/SPECS/<pkgname>/**/.*', and we want to extract
-          # the package name (ie the folder inside ./SPECS).
-          changed_pkgs=$(git diff-tree --diff-filter=d --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS/.*" || test $? = 1; } | sed -n 's#SPECS/\([^/]*\)/.*#\1#p' | sort -u | xargs)
-          changed_pkgs_extended=$(git diff-tree --diff-filter=d --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS-EXTENDED/.*" || test $? = 1; } | sed -n 's#SPECS-EXTENDED/\([^/]*\)/.*#\1#p' | sort -u | xargs)
+          # the package's directory name (ie the folder inside ./SPECS).
+          changed_pkg_dirs=$(git diff-tree --diff-filter=d --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS/.*" || test $? = 1; } | sed -n 's#SPECS/\([^/]*\)/.*#\1#p' | sort -u | xargs)
+          changed_pkg_dirs_extended=$(git diff-tree --diff-filter=d --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep "SPECS-EXTENDED/.*" || test $? = 1; } | sed -n 's#SPECS-EXTENDED/\([^/]*\)/.*#\1#p' | sort -u | xargs)
+
+          echo "Folders with modified files in this PR:"
+          echo "SPECS/*:          ${changed_pkg_dirs}"
+          echo "SPECS-EXTENDED/*: ${changed_pkg_dirs_extended}"
+
+          # For each package directory, get the names of all .spec files contained and add them to the list of SRPMS to repack.
+          # We need to filter since some directories may not contain .spec files we can rebuild.
+          changed_pkgs=""
+          for pkg in $changed_pkg_dirs; do
+            changed_pkgs="$changed_pkgs $(find SPECS/$pkg -name '*.spec' -exec basename {} \; | sed 's/\.spec//g' | xargs)"
+          done
+          changed_pkgs_extended=""
+          for pkg in $changed_pkg_dirs_extended; do
+            changed_pkgs_extended="$changed_pkgs_extended $(find SPECS-EXTENDED/$pkg -name '*.spec' -exec basename {} \; | sed 's/\.spec//g' | xargs)"
+          done
```

An idea to shorten things a bit:
```suggestion
          # the package's directory name (ie the folder inside ./SPECS).
          changed_pkg_dirs=$(git diff-tree --diff-filter=d --no-commit-id --name-only -r ${{ env.base_sha }} ${{ github.sha }} | { grep -oP "SPECS[^/]*/[^/]+" || [[ $? -eq 1 ]]; } | sort -u)

          echo "Folders with modified files in this PR:"
          echo "${changed_pkg_dirs}"

          # For each package directory, get the names of all .spec files contained and add them to the list of SRPMS to repack.
          # We need to filter since some directories may not contain .spec files we can rebuild.
          changed_pkgs=()
          while IFS= read -r pkg_dir; do
            changed_pkgs+=("$(find $pkg_dir -name '*.spec' -exec basename {} .spec \;)")
          done <<< "$changed_pkg_dirs"
```

You'd then also need to change

`echo "changed_pkgs=${changed_pkgs[@]}" >> $GITHUB_ENV`
to
`echo "changed_pkgs=${changed_pkgs}" >> $GITHUB_ENV.